android-agent/src/main/java/io/opentelemetry/android/internal/services/network/CurrentNetworkProvider.java

```diff
@@ -26,44 +30,37 @@
  * A provider of {@link CurrentNetwork} information. Registers itself in the Android {@link
  * ConnectivityManager} and listens for network changes.
  */
-public final class CurrentNetworkProvider {
+public final class CurrentNetworkProvider implements Startable {
 
-    static final CurrentNetwork NO_NETWORK =
+    public static final CurrentNetwork NO_NETWORK =
             CurrentNetwork.builder(NetworkState.NO_NETWORK_AVAILABLE).build();
-    static final CurrentNetwork UNKNOWN_NETWORK =
+    public static final CurrentNetwork UNKNOWN_NETWORK =
             CurrentNetwork.builder(NetworkState.TRANSPORT_UNKNOWN).build();
 
-    /**
-     * Creates a new {@link CurrentNetworkProvider} instance and registers network callbacks in the
-     * Android {@link ConnectivityManager}.
-     */
-    public static CurrentNetworkProvider createAndStart(Application application) {
-        Context context = application.getApplicationContext();
-        CurrentNetworkProvider currentNetworkProvider =
-                new CurrentNetworkProvider(NetworkDetector.create(context));
-        currentNetworkProvider.startMonitoring(
-                CurrentNetworkProvider::createNetworkMonitoringRequest,
-                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE));
-        return currentNetworkProvider;
+    public static CurrentNetworkProvider create(Application application) {
+        return new CurrentNetworkProvider(
+                NetworkDetector.create(application),
+                (ConnectivityManager) application.getSystemService(Context.CONNECTIVITY_SERVICE));
     }
 
     private final NetworkDetector networkDetector;
+    private final ConnectivityManager connectivityManager;
 
     private volatile CurrentNetwork currentNetwork = UNKNOWN_NETWORK;
-    private final List<NetworkChangeListener> listeners = new CopyOnWriteArrayList<>();
+    private final List<NetworkChangeListener> listeners =
+            new CopyOnWriteArrayList<>(); // visible for tests
```

Comment:
not actually visible tho....?

Reply:
if you're talking about the method that mutates the list, then maybe change to `// exposed thru method below for tests` or something

Reply:
Ah, my bad, I missed that comment when moving code around.

Reply:
I've just fixed it, turns out the comment belonged to a constructor right below this line.